mantidproject/mantid
# Use ArrayProperty instead of manually parsed string in `GetEiMonDet2`

This pull request updates [`GetEiMonDet`](http://docs.mantidproject.org/nightly/algorithms/GetEiMonDet-v2.html) to use the ArrayProperty for the *Detectors* property. Previously it was a string property with manual parsing.

**To test:**

1. `Load` some data, for example <mantid build directory>/ExternalData/Testing/Data/UnitTest/ILL/IN6/164192.nxs
2. Run [`FindEPP`](http://docs.mantidproject.org/nightly/algorithms/FindEPP-v1.html) on the workspace.
3. Test the [`GetEiMonDet`](http://docs.mantidproject.org/nightly/algorithms/GetEiMonDet-v2.html). *PulseInterval* for the above data is 6035us. The *Detectors* property should be able to handle input such as "11,12,15,100-110". The *IndexType* property should affect how *Detectors* (and *Monitor*) references the workspace. Note, that for the above data, monitors' detector IDs are 1001, 1002 and 1003.

Fixes #17702 

*Does not need to be in the release notes.*

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
